docs/blog/2019-11-10-how-to-convert-an-existing-gatsby-blog-to-use-mdx/index.md

```diff
@@ -0,0 +1,211 @@
+
+### Step 1
+
+Install [gatsby-plugin-mdx](/packages/gatsby-plugin-mdx/), the official plugin for using MDX with Gatsby. Also, install `@mdx-js/mdx` and `@mdx-js/react`.
+
+```
+npm install --save gatsby-plugin-mdx @mdx-js/mdx @mdx-js/react
+```
+
+### Step 2
+
+In `gatsby-config.js`, replace `gatsby-transformer-remark` with `gatsby-plugin-mdx`, along with the following changes:
+
+```diff:title=gatsby-config.js
+  {
+-  resolve: `gatsby-transformer-remark`,
++  resolve: `gatsby-plugin-mdx`,
+   options: {
++    extensions: [`.mdx`, `.md`],
+-    plugins: [
++    gatsbyRemarkPlugins: [
+       {
+         resolve: `gatsby-remark-images`,
+         options: {
+           maxWidth: 590,
+         },
+       },
+       {
+         resolve: `gatsby-remark-responsive-iframe`,
+         options: {
+           wrapperStyle: `margin-bottom: 1.0725rem`,
+         },
+       },
+       `gatsby-remark-prismjs`,
+       `gatsby-remark-copy-linked-files`,
+       `gatsby-remark-smartypants`,
+     ],
+   },
+  },
+```
+
+Then, add some options to `gatsby-plugin-feed`. This will allow the the RSS feed of the site to parse MDX.
+
+```diff:title=gatsby-config.js
+- `gatsby-plugin-feed`,
+
++ {
++   resolve: `gatsby-plugin-feed`,
++   options: {
++     query: `
++       {
++         site {
++           siteMetadata {
++             title
++             description
++             siteUrl
++           }
++         }
++       }
++     `,
++     feeds: [
++       {
++         serialize: ({ query: { site, allMdx } }) =>
++           allMdx.edges.map(edge =>
++             Object.assign({}, edge.node.frontmatter, {
++               description: edge.node.excerpt,
++               data: edge.node.frontmatter.date,
++               url: site.siteMetadata.siteUrl + edge.node.fields.slug,
++               guid: site.siteMetadata.siteUrl + edge.node.fields.slug,
++               custom_elements: [{ "content:encoded": edge.node.body }],
++             })
++           ),
++         query: `
++         {
++           allMdx(
++             limit: 1000,
++             sort: { order: DESC, fields: [frontmatter___date] },
++           ) {
++             edges {
++               node {
++                 fields { slug }
++                 frontmatter {
++                   title
++                   date
++                 }
++                 body
++               }
++             }
++           }
++         }
++         `,
++         output: `/rss.xml`,
++         title: `Gatsby RSS feed`,
++       },
++     ],
++   },
++ },
+```
+
+### Step 3
+
+In `gatsby-node.js`, change `allMarkdownRemark` to `allMdx` and `MarkdownRemark` to `Mdx`.
+
+```diff:title=gatsby-node.js
+# line 11
+- allMarkdownRemark(
++ allMdx(
+
+# line 35
+- const posts = result.data.allMarkdownRemark.edges
++ const posts = result.data.allMdx.edges
+
+# line 56
```

Comment:
Can we make line 56 a separate code snippet with its own explanation?

Reply:
done